### ISSUE TITLE
isBranchConflicted returns wrong information

### ISSUE THREAD

**Issue by NiasSt90** [renovatebot/renovate#18600]
### How are you running Renovate?

Self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

The function tries to reset to `baseBranch` only if `config.currentBranch !== baseBranch`.

````
    await git.reset(ResetMode.HARD);
    if (origBranch !== baseBranch) {
      await git.checkout(baseBranch);
    }
    await git.merge(['--no-commit', '--no-ff', `origin/${branch}`]);
````

But now take a look into prepareCommit(). It creates a new (local) branch and commits something onto it. 

````
    await gitRetry(() =>
      git.checkout(['-B', branchName, 'origin/' + config.currentBranch])
    );
````

**But it don't change `config.currentBranch` according to this.** 
Perhaps this was okay (i'm not sure , when it's the right time to change currentBranch). 

**But then isBranchConflicted() should always reset(HARD) and checkout(baseBranch).**

The problem occurs always for a second run (from cache.ts), after the first run do a branch update, 
because then the local git repository still stays on the last updated branch (from prepareCommit).

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Can you describe how this can be reproduced? e.g. 3-4 steps to perform?

**Comment by NiasSt90** [comment]
> Can you describe how this can be reproduced? e.g. 3-4 steps to perform?

it's difficult to see with a real repo but easy to see in the code.
 
First the repo should update a branch (prepareCommit was called) and has an outcome="automerged".

Then it will call itself again and before it called `await setBranchCache(branches);` 
from where all branches where checked for `isBranchConflicted`.
But the first `prepareCommit()` call created a local branch without changing the `config.currentBranch` variable.

An easy fix could be to always checkout the baseBranch. 
But perhaps the bug should be fixed better in prepareCommit. 
Because the missing change of `config.currentBranch` could trigger other side-effects.

**Comment by rarkins** [comment]
@NiasSt90 won't Renovate stop processing other branches after any automerge?

@zharinov could you check in case this is related to the work you're doing with the non-default branch automerge problem?

**Comment by NiasSt90** [comment]
> @NiasSt90 won't Renovate stop processing other branches after any automerge?

 no, from repository/index.ts

````
export async function renovateRepository(
  repoConfig: RenovateConfig,
  canRetry = true
): Promise<ProcessResult | undefined> {
....
      await instrument('onboarding', () =>
        ensureOnboardingPr(config, packageFiles, branches)
      );
      addSplit('onboarding');

      const res = await instrument('update', () =>
        updateRepo(config, branches)
      );
      setMeta({ repository: config.repository });
      addSplit('update');
      await setBranchCache(branches);**//<-- HERE THE BUG WAS TRIGGERED AND WRONG RESULT CACHED**
      if (res === 'automerged') { 
        if (canRetry) {
          logger.info('Renovating repository again after automerge result');
          const recursiveRes = await renovateRepository(repoConfig, false);
          return recursiveRes;
        }
        logger.debug(`Automerged but already retried once`);
      } else {
        await ensureDependencyDashboard(config, branches, packageFiles);
      }
      await finaliseRepo(config, branchList);
      // TODO #7154
      repoResult = processResult(config, res!);

````

**Comment by NiasSt90** [comment]
> @NiasSt90 We fixed a similar error recently could you confirm this is still an issue?

`git.prepareCommit()` still don't change the `config.currentBranch` state and therefore the error should still be triggered in `isBranchConflicted`.

**Comment by RahulGautamSingh** [comment]
It is the same issue, I will try to expain.
Before `prepareCommit` fn is called we checkout base branch and after the commit is made we again chekcout base branch. This way we remain on base branch before&after the commiting and pusshing tp an update branch.

I had accidentally removed the modified this logic-flow in a past merge and it was fixed recently. So you shouldn't observe it now.

Here are links to the previous issue and fix PR
 https://github.com/renovatebot/renovate/issues/17804
 https://github.com/renovatebot/renovate/pull/20658

**Comment by NiasSt90** [comment]
dup of #17804 and already fixed in #20658

**Comment by HonkingGoose** [comment]
Duplicate of #17804
